sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -1013,6 +1013,23 @@ pi_result OCL(piKernelSetExecInfo)(pi_kernel kernel,
   }
 }
 
+/// API to get the native handle of a PI object
+///
+/// \param handleType is an identifier representing the type of the handle
+/// \param piObject is the PI object to get the handle of
+/// \param nativeHandle is the native handle of piObject
+pi_result OCL(piGetNativeHandle)(pi_handle_type handleType, void *piObject,
+                                 pi_native_handle *nativeHandle) {
+  switch (handleType) {
+  case pi_handle_type::PI_NATIVE_HANDLE_MEM:
+  case pi_handle_type::PI_NATIVE_HANDLE_QUEUE:
+    *nativeHandle = reinterpret_cast<pi_native_handle>(piObject);
+    return PI_SUCCESS;
+  default:
+    return PI_INVALID_VALUE;
```

Comment:
why is this not handling other PI objects?

Reply:
I assume it easily could, but this was the minimum required to keep `get_mem` and `get_queue` on `interop_handler` working for OpenCL.

Reply:
OK, then maybe add a TODO comment about needing to implement others?

Reply:
@steffenlarsen ping

Reply:
I don't think this is relevant anymore.